### PR TITLE
[internal][trace] Support persisting `events`, `links` and `status.description`

### PR DESCRIPTION
**Pull request by zhengfeiwang**
# Description

This PR targets to persist some missing fields: `events`, `links` and `status.description`

**events and status.description**

image

**links**

image

# All Promptflow Contribution checklist:
- [x] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: src/promptflow/promptflow/_sdk/_service/apis/span.py [hunk 1]
```diff
@@ -70,7 +70,7 @@ def from_request() -> "ListSpanParser":
         SpanFieldName.END_TIME: fields.DateTime(dt_format=PFS_MODEL_DATETIME_FORMAT),
         SpanFieldName.STATUS: fields.Nested(status_model, skip_none=True),
         SpanFieldName.ATTRIBUTES: fields.Raw(required=True),
-        SpanFieldName.EVENTS: fields.List(fields.String),
+        SpanFieldName.EVENTS: fields.List(fields.Raw),
```
Comment (zhengfeiwang) [inline src/promptflow/promptflow/_sdk/_service/apis/span.py:73]: MARK: support strong types with model

File: src/promptflow/promptflow/_sdk/entities/_trace.py [hunk 2]
```diff
@@ -144,6 +162,8 @@ def _from_protobuf_object(obj: PBSpan, resource: typing.Dict) -> "Span":
         session_id = resource_attributes[SpanResourceAttributesFieldName.SESSION_ID]
         experiment = resource_attributes.get(SpanResourceAttributesFieldName.EXPERIMENT_NAME, None)
 
+        events = Span._from_protobuf_events(obj.events)
```
Comment (zhengfeiwang) [inline src/promptflow/promptflow/_sdk/entities/_trace.py:165]: TODO: add some tests to guard

File: src/promptflow/promptflow/_sdk/_service/apis/span.py [hunk 3]
```diff
@@ -57,6 +65,21 @@ def from_request() -> "ListSpanParser":
         SpanResourceFieldName.SCHEMA_URL: fields.String,
     },
 )
+event_model = api.model(
+    "Event",
```
Comment (wangchao1230) [inline src/promptflow/promptflow/_sdk/_service/apis/span.py:69]: may add some example value on event links in PR description

Comment (wangchao1230) [reply src/promptflow/promptflow/_sdk/_service/apis/span.py:69]: or add some test with such field

Comment (zhengfeiwang) [reply src/promptflow/promptflow/_sdk/_service/apis/span.py:69]: updated the pr desc

**Review by wangchao1230 — APPROVED** [review]

**Review by thy09 — APPROVED** [review]

**Pull request merged**